sdk/trace/src/test/scala/org/typelevel/otel4s/sdk/trace/SdkTraceScopeSuite.scala

```diff
@@ -0,0 +1,310 @@
+import cats.Functor
+import cats.effect.IO
+import cats.effect.IOLocal
+import cats.effect.LiftIO
+import cats.effect.MonadCancelThrow
+import cats.mtl.Local
+import munit.CatsEffectSuite
+import munit.ScalaCheckEffectSuite
+import org.scalacheck.Arbitrary
+import org.scalacheck.effect.PropF
+import org.typelevel.otel4s.sdk.context.Context
+import org.typelevel.otel4s.trace.SpanContext
+
+class SdkTraceScopeSuite extends CatsEffectSuite with ScalaCheckEffectSuite {
+
+  private implicit val spanContextArbitrary: Arbitrary[SpanContext] =
+    Arbitrary(Gens.spanContext)
+
+  test("current - return None when there is no SpanContext") {
+    for {
+      scope <- createTraceScope
+      current <- scope.current
+    } yield assertEquals(current, None)
+  }
+
+  test("rootScope - return None when there is no SpanContext") {
+    for {
+      scope <- createTraceScope
+      rootScope <- scope.rootScope
+      current <- rootScope(scope.current)
+    } yield assertEquals(current, None)
+  }
+
+  test("rootScope - use Context.root when there is no SpanContext") {
+    val value = "value"
+
+    for {
+      scope <- createTraceScope
+      _ <- assertIO(scope.current, None)
+
+      key <- Context.Key.unique[IO, String]("some-key")
+      ctxWithKey <- scope.contextReader(_.updated(key, value))
+
+      // the context wasn't modified yet, so it should be empty
+      _ <- assertIO(scope.contextReader(_.get(key)), None)
+
+      // here we use an explicit context with a custom key
+      _ <- scope.withContext(ctxWithKey) {
+        for {
+          _ <- assertIO(scope.contextReader(_.get(key)), Some(value))
+          rootScope <- scope.rootScope
+          _ <- assertIO(rootScope(scope.current), None)
+          _ <- assertIO(rootScope(scope.contextReader(_.get(key))), None)
+        } yield ()
+      }
+
+      rootScope <- scope.rootScope
+      _ <- assertIO(rootScope(scope.current), None)
+    } yield ()
+  }
+
+  test(
+    "rootScope - keep the current context when there is an invalid SpanContext"
+  ) {
+    val invalid = SpanContext.invalid
+    val value = "value"
+
+    for {
+      scope <- createTraceScope
+      _ <- assertIO(scope.current, None)
+
+      key <- Context.Key.unique[IO, String]("some-key")
+      ctxWithKey <- scope.contextReader(_.updated(key, value))
+
+      // the context wasn't modified yet, so it should be empty
+      _ <- assertIO(scope.contextReader(_.get(key)), None)
+
+      // here we use an explicit context with a custom key
+      _ <- scope.withContext(ctxWithKey) {
+        scope.noopScope {
+          for {
+            _ <- assertIO(scope.contextReader(_.get(key)), Some(value))
+            _ <- assertIO(scope.current, Some(invalid))
+            rootScope <- scope.rootScope
+            _ <- assertIO(rootScope(scope.current), Some(invalid))
+            _ <- assertIO(
+              rootScope(scope.contextReader(_.get(key))),
+              Some(value)
+            )
+          } yield ()
+        }
```

Comment:
I don't think this is what the Java backend does if you call `Tracer#noopScope`. we should probably make sure they have the same behaviour and also document that somewhere, but I that shouldn't hold up this PR

Reply:
Hm, the behavior is different indeed. 

I run the following code in oteljava:
```scala
for {
  sdk <- makeSdk()
  tracer <- sdk.provider.get("tracer")
  _ <- tracer.noopScope(
    sdk.local.get.flatMap(ctx => IO.println("ctx: " + ctx))
  )
} yield ()
```

The output is:
```
ctx: org.typelevel.otel4s.java.context.Context$Noop$@63c3ee1d
```

____

In OpenTelementry Java, the behavior is the following:
```scala
val key = ContextKey.named[String]("some-key")
val ctx = Context.root().`with`(key, "value").makeCurrent()

Context.current().get(key) == "value" // the context data is there

val noopScope = Span.wrap(SpanContext.getInvalid).makeCurrent()

Context.current().get(key) == "value" // the context data is still there

noopScope.close()
ctx.close()
```

____

I guess, we should propagate context data in `oteljava` too.